### DISCUSSION TITLE
Run detekt with all / different subset of rules?

### DISCUSSION THREAD

**Discussion by G00fY2** [detekt/detekt#3681] [Q&A] [answered]
Hi,

I hope I didn't miss anything in the docs or the Q&As. I would like to run detekt with different sets of rules.

**For example:** For pre-commit hooks I am e.g. only interested in `style` + `formatting` rules but on CI I'd like run detekt with `all` rules.

Is there any way to achieve that?

Thanks in advance!

**Comment by schalkms** [comment, marked as answer]
You can define the **common** rule configuration in one file and the **environment-specific** configuration in multiple others.
Detekt allows to pass multiple config files as seen [here](
https://detekt.github.io/detekt/cli.html). The suggestion would be to select the passed configuration files depending on the current environment.

```
--config, -c
      Path to the config file (path/to/config.yml). Multiple configuration 
      files can be specified with ',' or ';' as separator.
```

Does this help?

**Comment by cortinico** [comment]
> but on CI I'd like run detekt with all rules.

Adding a small side note: you can also use the `--all-rules` flag to enable all the rules on CI (you don't need two specific config files).
